realm/realm-annotations-processor/src/main/java/io/realm/processor/ClassMetaData.java

```diff
@@ -105,45 +113,44 @@ public boolean generate() {
 
         if (!categorizeClassElements()) return false;
         if (!checkListTypes()) return  false;
-        if (!checkMethods()) return false;
         if (!checkDefaultConstructor()) return false;
-        if (!checkRequiredGetters()) return false;
-        if (!checkRequireSetters()) return false;
+        if (!checkForFinalFields()) return false;
+        if (!checkForTransientFields()) return false;
+        if (!checkForVolatileFields()) return false;
 
         return true; // Meta data was successfully generated
     }
 
-    // Check that only allowed methods are present in the RealmObject class
-    private boolean checkMethods() {
-        for (ExecutableElement executableElement : methods) {
-            String methodName = executableElement.getSimpleName().toString();
-
-            // Check the modifiers of the method
-            Set<Modifier> modifiers = executableElement.getModifiers();
-            if (modifiers.contains(Modifier.STATIC)) {
-                continue; // We're cool with static methods. Move along!
-            } else if (!modifiers.contains(Modifier.PUBLIC)) {
-                Utils.error("The methods of the RealmObject class must be public", executableElement);
+    private boolean checkForTransientFields() {
+        for (VariableElement field : fields) {
+            if (field.getModifiers().contains(Modifier.TRANSIENT)) {
+                Utils.error("Transient fields are not allowed. Class: " + className + ", Field: " +
+                        field.getSimpleName().toString());
                 return false;
             }
+        }
+        return true;
+    }
 
-            // Check that getters and setters are valid
-            if (methodName.startsWith("get") || methodName.startsWith("is")) {
-                if (!checkGetterMethod(methodName)) {
-                    Utils.error(String.format("Getter %s is not associated to any field", methodName), executableElement);
-                    return false;
-                }
-            } else if (methodName.startsWith("set")) {
-                if (!checkSetterMethod(methodName)) {
-                    Utils.error(String.format("Setter %s is not associated to any field", methodName), executableElement);
-                    return false;
-                }
-            } else {
-                Utils.error("Only getters and setters should be defined in RealmObject classes", executableElement);
+    private boolean checkForVolatileFields() {
+        for (VariableElement field : fields) {
+            if (field.getModifiers().contains(Modifier.VOLATILE)) {
+                Utils.error("Volatile fields are not allowed. Class: " + className + ", Field: " +
+                        field.getSimpleName().toString());
                 return false;
             }
         }
+        return true;
+    }
 
+    private boolean checkForFinalFields() {
```

Comment:
that would be quite deceiving. People would think their final fields are persisted but they are really not.